core/templates/services/rte-helper-modal.controller.ts

```diff
@@ -218,7 +218,9 @@ export class RteHelperModalComponent {
       this.ngbActiveModal.dismiss(true);
     } else {
       this.ngbActiveModal.dismiss(false);
-      this.customizationArgsFormSubscription.unsubscribe();
+      if (this.customizationArgsFormSubscription) {
```

Comment:
Could you explain why this change was needed (what's the context where the subscription is null)? Perhaps worth adding a comment.

Reply:
The test `describe('when customization args has a valid youtube video', function ()` in the file [rte-helper-modal.controller.spec.ts](https://github.com/oppia/oppia/blob/0d07f468562521cfd398a2bee0f407cd85ed9a18/core/templates/services/rte-helper-modal.controller.spec.ts) was failing with the error `Cannot read property 'unsubscribe' of undefined` so I thought that it was because of the case when components are opened, then canceled without suffering alterations so I added thar condition. 

Now, after taking a better look into it I think the test was failing because the componentIsNewlyCreated should be set as true in the test since that is the case and right now is just undefined. Can I change the test to set componentIsNewlyCreated to true?

Reply:
@seanlip Sorry, I did that because the test `should close modal when clicking on cancel button` was failing and I thought there was indeed a case where the subscription is null when we change the video link and then cancel right after. After looking into it I realized I was mistaken and the problem was that the test should call `component.ngOnInit();`. Can I change the test to call this function?

Also, I think in the cancel function the line  `this.customizationArgsFormSubscription.unsubscribe();` should be outside of the if clause since the subscription is created even when the component is newly created. 

Do these changes seem okay?